docs/admin/cluster-management.md

```diff
@@ -14,47 +14,141 @@ certainly want the docs that go with that version.</h1>
+
+Tools in `cluster` folder are mainly `bash` scripts which make managing single cluster easier, but they are currently not designed to allow easy management of multiple clusters from a single machine. If it's necessary it is possible to replicate existing e2e test cluster setup that consists mainly of `hack/ginkgo-e2e.sh` script and special test configs for specific cloud providers e.g. `cluster/gce/config-test.sh`. This means that in general scripts from `cluster` directory find a running cluster by themselves, and you don't need to explicitly specify the cluster you want to modify, as those tools assume that there's the cluster running.
+
+Note the `cluster/kube-env.sh` script that defines the `KUBERNETES_PROVIDER` variable which is used by all other scripts. Default value is `gce` so if you want to manage non-GCE cluster you need to assign an identifier of your provider (e.g. `gke`, `aws`, `azure`, `vagrant`, etc.) to `KUBERNETES_PROVIDER` variable e.g. by putting the line
+```bash
+export KUBERNETES_PROVIDER=vagrant
+```
+to your `.bashrc` or `.profile` file.
 
 ## Upgrading a cluster
 
-The `cluster/kube-push.sh` script will do a rudimentary update; it is a 1.0 roadmap item to have a robust live cluster update system.
+GKE handles cluster upgrades for you, so you don't need to worry about it. If you already have a cluster running in some other environment and want to upgrade Kubernetes version running on it, you can use the `cluster/kube-push.sh` script. It works out of the box if your cluster runs on GCE and some other providers listed in `cluster` folder. Note that `kube-push.sh` requires that cloud providers implement necessary interface, thus not all of them are currently supported.
 
-## Updgrading to a different API version
+The upgrade is in-place, meaning that all user pods are kept alive throught this process. `kube-push.sh` has a few functionalities. It can either be used to upgrade all Kubernetes components to the currently built binaries (default option), or to do limited upgrade of either the master machine or a single node. Instead of upgrading to the head, you can also specify the version which you want to roll out.
 
-There is a sequence of steps to upgrade to a new API version.
+If you want to upgrade your whole cluster to, say, version 0.20.2 on a provider supported by `kube-push.sh` you use the command:
+```
+cluster/kube-push.sh v.0.20.2
+```
+In case when want to upgrade only running master to the version 0.20.2, you can use:
+```
+cluster/kube-push.sh -m v0.20.2
+```
+and if you want to upgrade a node `my_node` instead you write:
+```
+cluster/kube-push.sh -n my_node v0.20.2
+``` 
 
-1. Turn on the new api version
-2. Upgrade the cluster's storage to use the new version.
-3. Upgrade all config files. Identify users of the old api version endpoints.
-4. Update existing objects in the storage to new version by running cluster/update-storage-objects.sh
-3. Turn off the old version.
+## Scaling the running cluster
```

Comment:
I don't agree, as node.md holds only design information (for now), while this thing is about changing things (with example). I'll move it either way to make things go smooth.